src/main/host/descriptor/descriptor_table.rs

```diff
@@ -9,6 +9,7 @@ use crate::host::descriptor::{CompatFile, Descriptor};
 pub const FD_MAX: u32 = i32::MAX as u32;
 
 /// Map of file handles to file descriptors. Typically owned by a Process.
+#[derive(Clone)]
```

Comment:
I can't remember exactly what it does so I don't have any fix in mind.

Reply:
I started digging into how this might be worked around or fixed, but it'll definitely be a little involved.

I'm going to see if I can fix it, and move the tables from the `Process` to the  `Thread` at the same time. That'll exercise the fix (since the table will be shared across threads when `CLONE_FILES` is set, which it currently required) and let us support *not* sharing the table across threads (`CLONE_THREAD` without `CLONE_FILES`), though IDK if the latter happens in practice.

Reply:
#3082 removes `shutdown_helper`; will go ahead and merge this PR once that one is in.